tapir/shifts/models.py

```diff
@@ -721,6 +721,35 @@ def get_current_shift_exemption(self, date=None):
     def is_currently_exempted_from_shifts(self, date=None):
         return self.get_current_shift_exemption(date) is not None
 
+    def send_shift_reminder_emails(self):
+        today = datetime.date.today()
+        next_monday = today + datetime.timedelta((0 - today.weekday()) % 7)
+        next_sunday = next_monday + datetime.timedelta(days=7)
+        for attendance in ShiftAttendance.objects.with_valid_state().filter(
+            user=self.user,
+            slot__shift__start_time__gte=next_monday,
+            slot__shift__start_time__lte=next_sunday,
+        ):
+            self.send_shift_reminder_email(attendance.slot.shift)
+
+    def send_shift_reminder_email(self, shift: Shift):
```

Comment:
I feel this function should be idempotent so that we can re-run it as often as we want. This is useful e.g. if the cronjob failed halfway through, we fixed the problem, but want it to complete. We could have either a boolean or an Hstore with a key for this specific reminder email on the `ShiftAttendance`. This would make it flexible enough to accomodate more reminder emails (e.g. one ahead, or one after the shift) in the future.

Reply:
Good point!